src/controllers/bookController.ts

```diff
@@ -0,0 +1,128 @@
+import { NextFunction, Request, Response } from "express";
+import Book, { IBook } from "../models/Book";
+import { StatusCodes } from "http-status-codes";
+import { BadRequestError, NotFoundError } from "../errors";
+import mongoose from "mongoose";
+
+// Interface for query parameters
+interface IGetBooksQuery {
```

Comment:
Is there somewhere in the swagger doc comments in `bookRoutes.ts` we can list the optional query parameters allowed for a given route? The frontend team isn't going to know about these parameters unless they show up in the swagger docs.